tests/test_config_value.py

```diff
@@ -980,6 +1100,78 @@ def test_shear_value():
     np.testing.assert_almost_equal(sum1.getG1(), s.getG1())
     np.testing.assert_almost_equal(sum1.getG2(), s.getG2())
 
+    # Test NFWHaloShear
+    galsim.config.ProcessInput(config)
+    try:
+        # Raise an error because no world_pos
+        np.testing.assert_raises(ValueError, galsim.config.ParseValue, config, 'nfw', config,
+                                 galsim.Shear)
+        config['world_pos'] = galsim.PositionD(6,8)
+        # Still raise an error because no redshift
+        np.testing.assert_raises(ValueError, galsim.config.ParseValue, config, 'nfw', config,
+                                 galsim.Shear)
+        # With this, it should work.
+        config['gal'] = { 'redshift' : 1.3 }
+    except ImportError:
+        pass
+    nfw_halo = galsim.NFWHalo(mass=1.e14, conc=4, redshift=0.3)
```

Comment:
OK. I did these two, but I didn't try to comprehensively go through the tests to look for examples where doing this would be helpful.  If you notice any others that you think are egregious, let me know.  (Personally, I don't much mind magic numbers in tests -- in the main code is where I try to avoid them.)